cmd/Gopkg.toml

```diff
@@ -25,21 +25,21 @@
   revision = "f62e98d28ab7ad31d707ba837a966378465c7b57"
 
 [[constraint]]
-  branch = "release-1.9"
+  version = "kubernetes-1.10.1"
   name = "k8s.io/api"
 
 [[constraint]]
-  branch = "release-1.9"
+  version = "kubernetes-1.10.1"
   name = "k8s.io/apiextensions-apiserver"
 
 [[constraint]]
-  branch = "release-1.9"
+  version = "kubernetes-1.10.1"
   name = "k8s.io/apimachinery"
 
 [[constraint]]
-  branch = "release-1.9"
+  version = "kubernetes-1.10.1"
   name = "k8s.io/apiserver"
 
 [[constraint]]
-  branch = "release-6.0"
+  version = "kubernetes-1.10.1"
```

Comment:
I am curious why didn't we try `branch = "release-1.10"` in all the above. That will be better approach for updating the vendor in future because k8s doesn't follow semver style for versioning, otherwise default behavior of `dep` with version=v1.10.0 will achieve same benefits of `branch = "release-10"` 

Though, its not that big a deal, but wanted to know if it was deliberate.

Reply:
I have a habit of pinning to versions instead of branches because it means that `dep ensure` should be idempotent - I'm happy to use branches instead if that's what's desired. Note that the outer `Gopkg.toml` file pins to versions rather than branches, though.

Reply:
I think version pinning also works fine in this case. Re:`dep ensure` being idempotent, `dep` uses Gopkg.lock which locks these dependencies by package revisions.

About other Gopkg.toml, we are revisiting the decision to ship vendor.tar.gz etc, so will be experimenting with it today.